### PR TITLE
rails course & student app

### PR DESCRIPTION
**Pull request by franchizi**
Hey Ruben. Please review my course and students app. The main goal of this was for us to practice one to many relationships. I'm planning on updating the app with some of your earlier refactoring feedback also.

### CODE REVIEW

File: school/config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,17 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  get '/courses/new' => 'courses#new'
+  post '/courses' => 'courses#create'
+  get '/courses' => 'courses#index'
+  get '/courses/:id' => 'courses#edit'
+  post '/courses/edit' => 'courses#update'
+  get '/courses/delete' => 'courses#delete'
+  get '/courses/show' => 'courses#show'
```
Comment (ssrcruz) [inline school/config/routes.rb]: There is one small problem with your `show` and `delete` routes. When I try to view the course show page or try to delete a course I get a `NoMethodError` 
screen shot 2019-02-02 at 3 15 38 pm

Your route in this case is confused and trying to point to your edit action. 
If you want to follow the Rails convention of doing routes you can read up [here](https://guides.rubyonrails.org/routing.html)

How about trying something like

`get /contacts/:id  => contacts#show` display a specific contact <-- `:id` will be the contact id, it's so your route knows which contact to render on the show page.
`get /contacts/:id/edit => contacts#edit` <-- return an HTML form for editing a specific contact (edit.html.erb)
`PATCH/PUT  /contacts/:id => contacts#update` <-- pass in id to update a specific contact
`DELETE | /contacts/:id | contacts#destroy` <-- Your delete action expects an id that way it know which contact to delete

That way when you hit your actions you can access the contact id using `Course.find_by(id: params[:id])` in your controller, which you are already doing!

Comment (ssrcruz) [reply school/config/routes.rb]: Sorry, in your routes i meant `courses` not `contacts` lol

File: school/app/views/courses/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<a href='/courses/new'>Create New Course</a>
+
+<% @course.each do |course| %>
+<div>
+    <h4><%= course.name %></h4>
+    Location: <%= course.location %>
+    <p><%= course.description %></p>
+    <a href='/courses/<%= course.id %>'>Edit</a>
+    <a href='/courses/delete?id=<%= course.id %>'>Delete</a>
+    <a href='/courses/show?id=<%= course.id %>'>Show</a>
```
Comment (ssrcruz) [inline school/app/views/courses/index.html.erb]: Based on the feedback I gave you in your routes.rb file. 
You can now change the links on your form. If you type in `rails routes` you'll see all your possible routes. The [docs](https://guides.rubyonrails.org/layouts_and_rendering.html) has more detailed explanation.

**Comment by franchizi** [comment]
Thank you for the feedback!

**Pull request merged**